pkgs/build-support/gradle/update-deps.nix

```diff
@@ -0,0 +1,154 @@
+    export https_proxy="127.0.0.1:$MITM_CACHE_PORT"
+    export NIX_SSL_CERT_FILE="$MITM_CACHE_CA"
+    export SSL_CERT_FILE="$MITM_CACHE_CA"
+    SRC="$(mktemp -d)"
+    trap "rm -rf '$SRC'" EXIT
+    cp -r "$source/src" "$SRC/"
+    chmod -R +w "$SRC/src"
+    cd "$SRC/src"
+    source "$source/script2"
+    ${preBuild}
+    ${gradleCommands}
+    ${postBuild}
+  '';
+in
+writeShellScript "fetch-deps.sh" ''
+  set -eo pipefail
+  ${lib.optionalString (lib.hasInfix "$attrPath" code) ''
+    attrPath=${lib.escapeShellArg attrPath}
+    [[ -n "$attrPath" ]] || attrPath="$1"
+    [[ -n "$attrPath" ]] || attrPath="$UPDATE_NIX_ATTR_PATH"
+    [[ -n "$attrPath" ]] || (echo "Set UPDATE_NIX_ATTR_PATH or pass attr path as the first argument!" && exit 1)
+  ''}
+  outPath="${
+    # if this is an absolute path in nix store, use path relative to the output root
+    if lib.hasPrefix "/nix/store/" (toString depsPath)
+    then builtins.concatStringsSep "/" (lib.drop 4 (lib.splitString "/" (toString depsPath)))
+    # if this is an absolute path, just use that path
+    else if lib.hasPrefix "/" (toString depsPath)
+    then depsPath
+    # otherwise, use a path relative to the package
+    else ''$(nix-instantiate --eval -E "dirOf (with (${code}); meta.position)" | tr -d '"')/${depsPath}''
+  }"
+
+  source="$(nix-build --no-out-link -E "import ${./patched-source.nix} ({ self = ${code}; } // builtins.fromJSON "${
+    lib.escapeShellArg (builtins.toJSON (builtins.toJSON attrOverrides))
+  }")")"
+
+  source "$source/script1"
+  pushd "$(mktemp -d)"
+  MITM_CACHE_DIR="$PWD"
+  trap "rm -rf '$MITM_CACHE_DIR'" EXIT
+  ${openssl}/bin/openssl genrsa -out ca.key 2048
+  ${openssl}/bin/openssl req -x509 -new -nodes -key ca.key -sha256 -days 1 -out ca.cer -subj "/C=AL/ST=a/L=a/O=a/OU=a/CN=example.org"
+  export MITM_CACHE_PORT="''${mitmCachePort:-${toString mitmCachePort}}"
+  ${mitm-cache}/bin/mitm-cache \
+    -l"127.0.0.1:$MITM_CACHE_PORT" \
+    record \
+    --reject '\.(md5|sha(1|256|512:?):?)$' \
+    --forget-redirects-from '://(download\.savannah\.gnu\.org|github\.com|packages\.jetbrains\.team|sourceforge\.net:?)/' \
+    --forget-redirects-to '://(cdn\.jsdelivr\.net|plugins-artifacts\.gradle\.org:?)/' \
+    --record-text '/maven-metadata\.xml$' >/dev/null 2>/dev/null &
+  MITM_CACHE_PID="$!"
+  trap "kill '$MITM_CACHE_PID'" EXIT
+  sleep 0.5
+  ps -p "$MITM_CACHE_PID" >/dev/null || (echo "Failed to launch mitm-cache! Port $MITM_CACHE_PORT may be already used. Try setting "'$mitmCachePort' && exit 1)
+  export MITM_CACHE_CA="$PWD/ca.cer"
+  export MITM_CACHE_KS_PWD="$(head -c10 /dev/random | base32)"
+  echo y | ${jdk}/bin/keytool -importcert -file ca.cer -alias alias -keystore keystore -storepass "$MITM_CACHE_KS_PWD"
+  export MITM_CACHE_KEYSTORE="$PWD/keystore"
+  popd
+  ${preBuildNoSandbox}
+  # nsjail isn't necessary, it's only used to prevent messy build scripts from touching ~
+  ${nsjail}/bin/nsjail \
```

Comment:
`nsjail` is only supported on linux so prevents `.updateDeps` from building on darwin and other platforms. Would it be possible to conditionally enable the sandbox only for linux? Or could possibly call the nix daemon as a cross platform sandbox.